lib/config/options.rb

```diff
@@ -83,17 +83,7 @@ def reload_from_files(*files)
     end
 
     def to_hash
-      result = {}
-      marshal_dump.each do |k, v|
-        if v.instance_of? Config::Options
-          result[k] = v.to_hash
-        elsif v.instance_of? Array
-          result[k] = descend_array(v)
-        else
-          result[k] = v
-        end
-      end
-      result
+      __convert_to_hash(marshal_dump)
```

Comment:
Happy to move it back if you want. I initially thought we could use `__convert_to_hash` directly, but  ran into the symbol vs string problem.